File: src/main/java/com/treserve/event/Event.java
package com.treserve.event;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.treserve.user.User;
import com.treserve.venue.Venue;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.Instant;

@Entity
@Table(name = "events")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "venue_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Venue venue;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Column(name = "age_restriction", length = 10)
    private String ageRestriction;

    @Column(length = 50)
    private String category;

    @Column(name = "duration_minutes")
    private Integer durationMinutes;

    @Column(name = "start_time", nullable = false)
    private Instant startTime;

    @Column(name = "base_price", nullable = false, precision = 10, scale = 2)
    private BigDecimal basePrice;

    @Column(length = 20)
    @Builder.Default
    private String status = "ACTIVE";

    @Version
    private Long version;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    @JsonIgnore
    private User createdBy;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Instant updatedAt;
}

File: frontend/src/app/models/event.ts
export interface EventSession {
  id: number;
  eventId: number;
  startsAt: string;
  price: number;
}

export interface EventItem {
  id: number;
  title: string;
  description: string;
  venue: {
    id: number;
    name: string;
    address: string;
  };
  imageUrl: string | null;
  category: string;
  ageRestriction: string;
  startTime: string;
  basePrice: number;
}

export interface AdminEventSummary {
  id: number;
  title: string;
  venue: string;
  nextSession: string;
}

export interface VenueSeatTemplate {
  id: number;
  row: number;
  column: number;
  label: string;
  disabled: boolean;
}
